mlflow/utils/file_utils.py

```diff
@@ -328,12 +329,10 @@ def _copy_file_or_tree(src, dst, dst_dir=None):
     if dst_dir is not None:
         dst_subpath = os.path.join(dst_dir, dst_subpath)
     dst_path = os.path.join(dst, dst_subpath)
-
-    dst_dirpath = os.path.dirname(dst_path)
-    if not os.path.exists(dst_dirpath):
-        os.makedirs(dst_dirpath)
-
     if os.path.isfile(src):
+        dst_dirpath = os.path.dirname(dst_path)
+        if not os.path.exists(dst_dirpath):
```

Comment:
This was just moved down into if-branch. Previously this was done even if we were copying the whole tree which is unnecessary and shutils on windows don't like it. It would fail with path already exists.